game-app/game-core/src/main/java/games/strategy/engine/framework/ClientGame.java

```diff
@@ -218,7 +219,7 @@ public void addChange(final Change change) {
   }
 
   @Override
-  public IRandomSource getRandomSource() {
+  public @Nullable IRandomSource getRandomSource() {
     return null;
```

Comment:
I would guess a "OperationNotSupportedException" might be more appropriate here.